src/potions.ts

```diff
@@ -417,3 +430,134 @@ export function bathroomFinance(embezzlers: number): void {
+}
+
+class VariableMeatPotion {
+  potion: Item;
+  effect: Effect;
+  duration: number;
+  softcap: number; // Number of turns to cap out variable bonus
+  meatBonusPerTurn: number; // meat% bonus per turn
+  cappedMeatBonus: number;
+
+  constructor(
+    potion: Item,
+    softcap: number,
+    meatBonusPerTurn: number,
+    duration?: number,
+    effect?: Effect
+  ) {
+    this.potion = potion;
+    this.effect = effect ?? effectModifier(potion, "Effect");
+    this.duration = duration ?? numericModifier(potion, "Effect Duration");
+    this.softcap = softcap;
+    this.meatBonusPerTurn = meatBonusPerTurn;
+    this.cappedMeatBonus = softcap * meatBonusPerTurn;
+  }
+
+  use(quantity: number): boolean {
+    return use(quantity, this.potion);
+  }
+
+  price(historical: boolean): number {
+    // If asked for historical, and age < 14 days, use historical.
+    // If potion is not tradeable, use retrievePrice instead
+    return this.potion.tradeable
+      ? historical && historicalAge(this.potion) < 14
+        ? historicalPrice(this.potion)
+        : mallPrice(this.potion)
+      : retrievePrice(this.potion);
+  }
+
+  getOptimalNumberToUse(yachtzees: number, embezzlers: number): number {
+    const barfTurns = Math.max(0, estimatedTurns() - yachtzees - embezzlers);
+
+    const potionAmountsToConsider = [] as number[];
+    for (const fn of [Math.floor, Math.ceil]) {
+      for (const sc of [0, this.softcap]) {
+        for (const em of [0, embezzlers]) {
+          for (const bt of [0, barfTurns]) {
+            potionAmountsToConsider.push(fn((yachtzees + em + bt + sc) / this.duration));
+          }
+        }
+      }
+    }
+
+    const profits = [] as number[];
+    const profitsMap = new Map(
+      potionAmountsToConsider.map((n) => {
+        const profit = this.valueNPotions(n, yachtzees, embezzlers, barfTurns);
+        profits.push(profit);
+        return [profit, n];
+      })
+    );
+
+    const bestProfits = profits.reduce((a, b) => (a > b ? a : b));
+    if (bestProfits > 0) {
+      const nPotionsToUse =
+        potionAmountsToConsider[profitsMap.get(bestProfits) ?? 0] -
+        Math.floor(haveEffect(this.effect) / this.duration);
+      return Math.max(nPotionsToUse, 0);
+    }
+    return 0;
+  }
+
+  valueNPotions(n: number, yachtzees: number, embezzlers: number, barfTurns: number): number {
+    const yachtzeeValue = 2000;
+    const embezzlerValue = baseMeat + 750;
+    const barfValue = (baseMeat * turnsToNC) / 30;
+
+    const totalCosts = n * this.price(false);
+    const totalDuration = n * this.duration;
+    let cappedDuration = Math.max(0, totalDuration - this.softcap + 1);
+    let decayDuration = Math.min(totalDuration, this.softcap - 1);
+    let totalValue = 0;
+    const turnTypes = [
+      [yachtzees, yachtzeeValue],
+      [embezzlers, embezzlerValue],
+      [barfTurns, barfValue],
+    ];
+
+    // eslint-disable-next-line prefer-const
+    for (let [turns, value] of turnTypes) {
```

Comment:
We only need to use `let` in for loops of the form `for (let i = start; i <= end; i++)` or similar things. `const` works just fine for for...of loops. Or for...in loops, but those are dumb.

Reply:
I need to decrement turns (although value is left untouched), so I do need a `let` somewhere